File: FoundBuddy-backend/src/main/java/com/example/foundbuddybackend/repository/ItemRepository.java
package com.example.foundbuddybackend.repository;

import com.example.foundbuddybackend.model.Item;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

/**
 * JPA repository for {@link Item} entities.
 */
@Repository
public interface ItemRepository extends JpaRepository<Item, Long> {

    /**
     * Finds all items whose title or description contains the given substring
     * (case‑insensitive).  Spring Data JPA automatically derives the query
     * implementation based on the method name.
     *
     * @param title fragment of the title to search for
     * @param description fragment of the description to search for
     * @return a list of matching items
     */
    List<Item> findByTitleContainingIgnoreCaseOrDescriptionContainingIgnoreCase(String title, String description);
}

File: FoundBuddy-frontend/app/src/main/java/com/example/foundbuddy/model/ItemRepository.kt
package com.example.foundbuddy.model

object ItemRepository {
    private val itemsList = mutableListOf<Item>()

    val items: List<Item> get() = itemsList.toList()

    fun search(query: String): List<Item> {
        return itemsList.filter {
            it.title.contains(query, ignoreCase = true) || it.description.contains(query, ignoreCase = true)
        }
    }

    fun addItem(item: Item) {
        itemsList.add(item)
    }
}
